internal/service/worker.go

```diff
@@ -379,8 +379,11 @@ func (w *Worker) extractToken(endpoint string) (string, error) {
 // fetchAnnotations is used to get the annotations based on a token and preprocess them. The second return parameter is
 // a cleanup function that always need to be executed once the information is no longer needed. The cleanup function is
 // only available in case there is no errors.
-func (w *Worker) fetchAnnotations(ctx context.Context, token string, page int) ([]any, func(), error) {
-	annotations, err := w.AnnotationStorage.FetchAnnotation(ctx, token)
+func (w *Worker) fetchAnnotations(ctx context.Context, token string, page int) (annotations []any, cleanup func(), err error) {
+	span, ctx := ddTracer.StartSpanFromContext(ctx, "Worker.fetchAnnotations")
+	defer func() { span.Finish(ddTracer.WithError(err)) }()
```

Comment:
You're missing the `err` here, usually this requires the return to use a named return `(_ []any, function(), err error)`.